packages/react/src/components/Button/index.tsx

```diff
@@ -1,42 +1,40 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { ButtonHTMLAttributes, forwardRef, ReactNode } from 'react';
 import classNames from 'classnames';
 
-export interface ButtonProps
-  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
   variant?: 'primary' | 'secondary' | 'error' | 'link';
-  buttonRef?: React.Ref<HTMLButtonElement>;
-  disabled?: boolean; // todo, investigate why disabled is needed
   thin?: boolean;
+  startIcon?: ReactNode;
+  endIcon?: ReactNode;
 }
 
-export default class Button extends React.Component<ButtonProps> {
-  static defaultProps = {
-    variant: 'primary',
-    buttonRef: () => {}
-  };
-
-  static propTypes = {
-    variant: PropTypes.oneOf(['primary', 'secondary', 'error', 'link']),
-    children: PropTypes.node,
-    className: PropTypes.string,
-    buttonRef: PropTypes.oneOfType([
-      PropTypes.func,
-      PropTypes.shape({ current: PropTypes.any })
-    ])
-  };
-
-  static displayName = 'Button';
-
-  render() {
-    const {
-      variant,
+const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      variant = 'primary',
+      thin,
       children,
       className,
-      buttonRef,
-      thin,
+      startIcon,
```

Comment:
I think we should favor composition over `propTypes` here. The feels way too prescriptive. The icon should naturally be placed in the button where it's needed:

```jsx
<Button>
  <Icon type="icon-type">
  Submit
</Button>
```

or

```jsx
<Button>
  Submit
  <Icon type="icon-type">
</Button>
```

It would be better to have inheritance styles [here](https://github.com/dequelabs/cauldron/blob/develop/packages/styles/button.css) so the icon would naturally inherit styles from the button. You're already on the right track with setting `flex` on the button, just need the appropriate styles for the icon.

```css
.Button--primary .Icon,
.Button--secondary .Icon,
.Button--clear .Icon,
.Button--error .Icon {
  /* icon styles */
}
```